tests/integration/e2e/RoomAdmin.test.tsx

```diff
@@ -369,4 +369,108 @@ describe('End-to-end Room Admin updating', function () {
       });
     });
   });
+  describe('Editing a room', function () {
+    let renderResult: RenderResult;
+    let roomNameInput;
+    let capacityInput;
+    beforeEach(async function () {
+      renderResult = render(
+        <MemoryRouter initialEntries={['/room-admin']}>
+          <App />
+        </MemoryRouter>
+      );
+      await renderResult.findAllByText(rooms[0].building, { exact: false });
+    });
+    context('when the modal opens', function () {
+      const testRoom = rooms[0];
+      beforeEach(async function () {
+        const editButton = renderResult.getByLabelText(`Edit room information for ${testRoom.building} ${testRoom.name}`);
+        fireEvent.click(editButton);
+        await renderResult.findByText('Note: * denotes a required field', { exact: false });
+      });
+      it('shifts the focus to the modal header', function () {
+        strictEqual(document.activeElement.textContent, `Edit room ${testRoom.building} ${testRoom.name}`);
+      });
+      context('when the field values provided are invalid', function () {
+        beforeEach(function () {
+          const modal = renderResult.getByRole('dialog');
+          roomNameInput = within(modal).getByLabelText('Room Number', { exact: false }) as HTMLInputElement;
+          capacityInput = within(modal).getByLabelText('Capacity', { exact: false }) as HTMLInputElement;
+        });
+        context('when Room Name is not provided', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(roomNameInput,
+              { target: { value: '' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Room number is required to submit this form';
+            return waitForElement(
+              () => renderResult.getByText(errorMessage, { exact: false })
+            );
+          });
+        });
+        context('when Capacity is not provided', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(capacityInput,
+              { target: { value: '' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Capacity is required to submit this form, and it must be a positive number';
+            return waitForElement(
+              () => renderResult.getByText(errorMessage, { exact: false })
+            );
+          });
+        });
+        context('when Capacity is invalid', function () {
+          it('displays a validation error', async function () {
+            fireEvent.change(capacityInput,
+              { target: { value: '100z' } });
+            const submitButton = renderResult.getByText('Submit');
+            fireEvent.click(submitButton);
+            const errorMessage = 'Capacity is required to submit this form, and it must be a positive number';
```

Comment:
Don't need "positive"

Reply:
Also updated this in [c345f15](https://github.com/seas-computing/course-planner/pull/581/commits/c345f15902f6fb3fe68bce21766aa084e45a7520).